packages/webview_flutter/webview_flutter_web/test/webview_flutter_web_test.dart

```diff
@@ -64,7 +65,11 @@ void main() {
       // Run
       controller.loadHtmlString('test html');
       // Verify
-      verify(mockElement.src = 'data:text/html,${Uri.encodeFull('test html')}');
+      verify(mockElement.src = Uri.dataFromString(
+        'test html',
```

Comment:
These test cases don't actually exercise the problem described in your issue. Would either of these even fail if you reverted the non-test changes?

Reply:
The tests wouldn't pass before this PR's changes because instead of starting the `src`'s value with `data:text/html`, it now starts with `data:text/html;charset=utf-8`.

It would be more accurate for the test to only check if the `mockElement.src` starts with `data:text/html;charset=utf-8`, I guess. But I don't know how to do it with mock tests or if it's even necessary.

Reply:
That test failure would have nothing to do with the issue you fixed. I could make these tests pass without fixing the bug by just adding `;charset=utf-8` into the interpolated string.

The issue you are fixing is about properly escaping special characters--specifically `#` in your examples--so that's what the test should be verifying.

Reply:
(In fact, the changelog should say "incorrect escaping", not "unreliable encoding", since this is about escaping rather than encoding.)

Reply:
You're right I'll fix the test to include the new escaping. But should it be renamed to only fix escaping even though it adds the `;charset=utf-8`?

Reply:
If you want to also mention adding explicit encoding as a secondary change, feel free. But there have been zero reports of encoding issues so it's not particularly important to mention.